---
title: "DepMap Explorative Data Analysis (EDA)"
author: "Yamila Timmer"
date: '`r Sys.Date()`'
output:
  html_document:
    toc: true
    toc_depth: '3'
    number_sections: true

lang: en
bibliography: references.bib
---


```{r load-libraries, message = FALSE}
library(shiny)
library(ggplot2)
library(tidyr)
library(plotly)
library(naniar)
```


# Data Exploration

## Importing data

**19-11-2024**

In order to analyse the data, I first have to import the needed data files. The two data files I will download are:

### Model.csv
- Contains metadata describing all cancer models which are referenced within the DepMap portal, but in this case specifically 'OmicsExpressionProteinCodingGenesTPMLogp1.csv'. The metadata is very extensive and descriptions of each of the metadata columns can be found in the appendix of this EDA. The descriptions are copied from the documentation of the DepMap database.

### OmicsExpressionProteinCodingGenesTPMLogp1.csv

- Contains the gene expression transcripts per million (TPM) values of the protein coding genes for DepMap cell lines as described in `model.csv`. TPM is a way of normalization of transcript counts data in measuring gene expression levels. Normalization is necessary, because the number of reads will differ per cell line during the RNA sequencing step. TPM measures the amount of mapped reads per million mapped reads, which results in a display of ratio in mapped reads instead of an absolute number, thus normalizing the data. Figure 1 displays the corresponding formula that belongs to normalization using TPM [@Zhao2020].

- Gene expression values are obtained through RNA-sequencing, using RSEM, which is a transcript counting tool. Specifically `unstranded mode` was used. Finally a log2 transformation was performed, which scales the data according to a log2 scale, reducing distance between values, and especially outliers. In this a pseudo-count of 1 was used, which results in the following formula: log2(TPM+1). The plus 1 as pseudo counts results in there being no negative values in the final data.

![Formula for normalization using `transcripts per million` (TPM)](../media/tpm.png)

## Loading the data

In the following two code blocks I read in both .csv-files and save them as variables. I use read.csv to do so, because of the corresponding file type. Using na.strings I convert all "", "unknown" and "Unknown" values to NA values so all missing values have an equal value of NA.

```{r load-data}
setwd("C:\\Users\\yamil\\OneDrive - Hanze\\Bio-informatica\\Jaar 2\\2.1 applicatie\\goede_git\\") # set working directory

model <- read.csv("Model.csv", na.strings = c("", "unknown", "Unknown"))
expression_db <- read.csv("../OmicsExpressionProteinCodingGenesTPMLogp1.csv")
```

## First look at the datasets

- Model.csv contains `r nrow(model)` rows and `r ncol(model)` columns.
- OmicsExpressionProteinCodingGenesTPMLogp1.csv contains `r nrow(expression_db)` rows and `r ncol(expression_db)` columns.

For a further look at the expression_db, the data will need to made tidy (see: chapter "Making the data tidy"). However for model.csv this is not the case, thus I will first be taking a look at this data set. First, I would like to take a look at the sex distribution of the patients that the cancer cell lines have been retrieved from, to see if there is a chance that there is any sort of bias here.

```{r determine-sex-bias-barplot}
ggplot(data = model, 
       aes(x = Sex, 
           fill = Sex)) +
  geom_bar(stat = "count") +
    theme_minimal()
```

From the figure above it can be seen that the male sex is dominant in this data set (n = `r nrow(model[model$Sex == 'Male', ])`), there seems to be around 25% more biologically male patients than there are biologically female patients (n = `r nrow(model[model$Sex == 'Female', ])`). This sex bias could lead to problems, for example when, for example the user of the application notices that most cell lines in specific cancer types are mostly male. This could lead to the user interpreting this as that mostly men suffer from that type of cancer, but this could just be a coincidence as most of the patients were male. In order to prevent such scenarios from happening, I think it's important that I display metadata in the app, also showing the total number of patients per sex, but also per age category, race, etc. (all the filtering options). 

Furthermore it seems that for `r nrow(model[model$Sex == 'NA',])` patients the sex is unknown, this could be due to errors during the research, where the sex did not get registered either by human mistakes or problems with the database/system. There were also values with 'unknown', but I converted all of those to NA, for easier handling during this EDA. In the application I will convert all NA's back to actual values (unknown), because I still want to give the user the option to include the unknown values. Seeing as a factor like sex might not always be important for cancer research. 


```{r determine-race-bias-barplot}
ggplot(data = model, 
       aes(x = PatientRace, 
           fill = PatientRace)) +
  geom_bar(stat = "count") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = -90))
    
```

From the figure above it can be seen that the distribution of races is not very divided, most patients did not get their race registered, which results in an 'NA' value (n = `r nrow(model[model$PatientRace == 'NA', ])`). Other than that, Caucasian seems to be the second most prominent race (n = `r nrow(model[model$PatientRace == 'caucasian', ])`), next is Asian (n = `r nrow(model[model$PatientRace == 'asian', ])`) and after this comes black/African american (n = `r nrow(model[model$PatientRace == 'black_or_african_american', ])`) and east Indian (n = `r nrow(model[model$PatientRace == 'east_indian', ])`). The other races do not seem to be very represented, as they are not at all visible in the plot. Even though so much race data is missing, I still want to include it in the metadata filters for the application. A user might for example want to study effects of cancer in specifically Caucasian or Asian patients (or even black/African american or east Indian patients, however there is not much data to work with there). 


## Making the data tidy

### Preparing expression_db

**26-11-2024**

Before moving on with making the expression_db tidy, some minor changes will be made to the column names, in order to make future processes easier. First, the name of the 'X' column in expression_db is changed to 'ModelID', to make the merging process easier (see chapter: merging the data sets).

```{r change-column-name}
colnames(expression_db)[1] <- "ModelID" # Change column 'X' to 'ModelID'
```

**19-11-2024**

Second, using regex I will change the gene column names. Reason being that the column names as of now don't lend themselves for an easy user-experience, due to the names being confusing. I believe that having just the gene name itself is enough for the user of the application to achieve what he or she wants. 

An example of the gene column name: '*TSPAN6..7105.*', instead of just 'TSPAN6'. All of the gene names follow a similar pattern to '..numbers.', with a regex statement this pattern can easily be removed, which results in more 'clean' names for all the genes. 

I use the regex statement "`\\.\\.[0-9]+\\.", '\\.`". `\\.` stands for a singular '.' in the pattern, the backslashes are needed because '.' can also be interpreted as 'any symbol'. [0-9] indicates any number from 0-9 and the '+' indicates that at least 1 number is expected, but that it will still be recognized even if there is 2 or more numbers in a row. Using the sub()-function I substitute this regex statement with "", meaning it will fully be removed. Resulting in column names with cleaner versions of the gene names.

```{r simplify-gene-names}
# converts gene names of columns to simplified, cleaner version using regex statement
for (col in 2:ncol(expression_db)){
    colnames(expression_db)[col] <-  sub("\\.\\.[0-9]+\\.", "", colnames(expression_db)[col])
}
```


**26-11-2024**

`expression_db` is formatted in a way that it is not 'clean' data. Meaning it does not meet the Tidy requirements. The dataframe contains a seperate column for each gene and within each column it has the expression levels per row (meaning per tumor cell line). However the preferred, or Tidy way, is to have one column per variable, meaning that all gene columns should be combined into one 'gene' column and all expression columns should be combined into one 'expression' column. This can be achieved using the tidyr function `pivot_longer()`. The gene columns are selected by giving a vector which contains the column names. 

The index 2:ncol(expression_db) selects all columns starting from the second column, which is where the gene columns start (column 1 contains the `ModelID`). The gene columns are converted to one gene column, named '*gene*' and all expression values are converted to a seperate column, named '*expression*'.

```{r make-data-tidy}
tidy_expression <- expression_db %>% 
  pivot_longer(
    cols = 2:ncol(expression_db),
    names_to = "gene",
    values_to = "expression"
  )
```


The metadata in the `model` data frame is already formatted according the Tidy requirements, seeing as all columns have a unique variable, so no changes will have to be made to this data frame.

## Look at expression

Now that the expression data is tidy, we can now see what this data looks like. First I will be visualizing the ranges of the expression data using a box plot.

```{r filter-data}
filtered_metadata <- model %>%
    filter(Sex %in% c("Male", "Female"))
filtered_expr <- tidy_expression %>%
            filter(gene %in% c("TSPAN6", "DPM1"))

```

```{r merge-data}
merged <- merge(filtered_metadata, filtered_expr, by = "ModelID", all = FALSE)

```

```{r generate-boxplot}

  ggplot(data = merged,
         aes(x = gene, 
             y = expression,
             fill = Sex)) +
    geom_boxplot() +
    labs(y = "Expression level(log2 TPM)") + 
    theme_minimal() +
    theme(axis.text.x = element_text(angle = -90)) # Rotate gene names x-axis

```

Boxplots are a good way to show the range of expression and to visualise within what borders most values lie. Above can be seen that the average expression for gene DPM1 is the highest in women. The same goes for expression of TSPAN6.


## Merging the datasets

**19-11-2024**

In the application itself, I will not be merging the two data sets beforehand, seeing as that would add a lot of unnecessary extra metadata when converting the merged data to the long (tidy) format. Which will result in long loading times when launching the application. Hence I made it so that the two data sets will be merged after the filtering step, right before generating the plot. By doing this there will be a lot less data that has to be merged, because the data is filtered and is a lot less rows than the unfiltered data. The two data sets will be merged in the server script for the application, with the following code:

```{r merge-data-application}
merged <- merge(filtered_metadata[, c("ModelID", "StrippedCellLineName")], filtered_expr, by = "ModelID", all = FALSE)
```

The code merges the data sets `filtered_metadata` (which is the filtered result of model.Rdata, containing the metadata) and `filtered_expr` (which is the filtered result of tidy_expression which contains the gene expression data). The two data sets are merged on a corresponding factor, namely the ModelID. The ModelID is given in a column called `X` in `filtered_expr` and a column called `ModelID` in `filtered_metadata`. Using the merge()-function in base R, I give both tables and the columns on which they should be matched on and thereafter merged. 

`all = FALSE` tells the function to only keep the rows in which a match and merge has taken place, meaning that any of the rows from `model` or `expression_db` without a matching ID, will be removed. The rows that will be removed will only be rows from `model`, as not all cell lines from model have been used in expression_db. 

## Saving the data
In order to load the data in the application, without having to go through all the steps of cleaning up the data again, I'm saving `model` and `tidy_expression` as R.data objects.

```{r save-data}
save(model, file = "C:\\Users\\yamil\\OneDrive - Hanze\\Bio-informatica\\Jaar 2\\2.1 applicatie\\model.Rdata")
save(tidy_expression, file = "C:\\Users\\yamil\\OneDrive - Hanze\\Bio-informatica\\Jaar 2\\2.1 applicatie\\tidy_expression.Rdata")

```


## Visualising missing data

**27-11-2024**

The figure below is generated using the Naniar package, which offers tools to visualize missing data. It can easily be told from the figure below that (almost) all values are missing in the `TissueOrigin`, `StagingSystem`, `PatientTumorGrade` and `DataSharedIndbGaP` columns. Even though the figure displays the amount of missing values as 100%, it cannot be certainly said that ALL values are missing, because of the size of the data frame, so even if some rows do have these values, it will not show up in the figure. `WTSIMasterCellID` and `COSMICID` seem to have missing values in around half of the rows (a bit over 50%). And `Age` seems to be missing values in approximately 18.75% of the rows. All other variables seem to have a very low number of missing values, which makes these variables a good option as metadata to use for filtering in the application by the user. However it seems there are a lot more missing values, given as "", instead of 'NA'. These need to be converted properly, so I have added '`na.strings=""`' to the read.csv statement for model.csv. Now it seems clear that the amount of missing data is a lot larger than I first thought, as can be seen in figure. Other, not yet named columns that also seem to be missing (almost) all data are: `PatientTreatmentType`, `PatientTreatmentDetails`,`ModelDerivationMaterial`, `HCMIID`, `PatientTreatmentResponse`, `Stage`, `EngineeredModel`, `PatientTreatmentStatus`, `PublicComments`, `PatientMolecularSubtype`, `LegacyMolecularSubtype`. The following columns seem to be missing around 25-60% of the data: `PatientRace`, `LegacySubSubtype`, `SangerModelID`, `CatalogNumber`. Based on this data it will have to be carefully evaluated which metadata columns are suitable for allowing the user to filter the data.

```{r visualise-missing-data}
gg_miss_var(model, show_pct = TRUE) + labs(y = "Missing values (%)")
```

# Visualization
## Barplot

**20-11-2024**

A simple, yet efficient way of visualizing expression values and cell lines can be done using a bar plot. For this I am using the ggplot library, with the geom_bar function. For the data I give the merged data frame, with the gene (in this case, TSPAN6) as x-axis and the tumor cell line name as y-axis. This results in expression levels being shown horizontally (x-axis) and corresponding cell line names vertically (y-axis). I ordered the expression levels from high to low, seeing as in cancer therapy one might be looking for highly over expressed genes, in order to develop a treatment that could interfere on the expression of that gene. I sorted them using the reorder()-function, by first giving the `StrippedCellLineName` column, and then telling it to sort it based off of the expression values in the gene (TSPAN6). 

However one might also be interested in genes with a very low expression, because this gene could also possibly be involved in the development of cancer in that cell line. The gene can for example be a tumor suppressor gene, which can also possibly be treated if a medication would be developed that could increase the expression of that gene again. I am planning on allowing the user to sort from either high-low or low-high for gene expression levels in the application. Sorting from low-high can be done by adding a '-' in the following line: `y=reorder(StrippedCellLineName,-TSPAN6)`. The code below is just a first visualization, later I will add onto the code and write it as a function so it can easily be used with different parameters, which will be needed for the application.

As of now I am only using the head of the data, as there are a lot of tumor cell lines (1517), by using various filtering methods I would like to lower this amount, for example let the user filter between specific expression levels, or only include cell lines extracted from certain types of tumors, so that the application won't have to display such an extensive bar plot.

```{r}
# Generates horizontal barplot of chosen gene and cell lines that express this gene 
ggplot(data = head(merged), aes(x = expression,
                              y = StrippedCellLineName)) + 
  geom_bar(stat = "identity", fill = 'blue') + 
  ylab("Tumor Cell Line") + 
  xlab("Expression level (log2 TPM)") + 
  theme_minimal() +
    facet_wrap(~ gene)

```

From the figure above can be seen that appareantly the gene 'DPM1' is expressed very strongly in the 3 celllines U937, PLB985 and MONOMAC6. However, the gene TSPAN6 seems to have very low expression in these same genes.



# Developing the application


## Thinking about what should be in the application

**21-11-2024**

There is a lot of metadata available for these data sets. In the chapter 'Importing data', all metadata variables can be seen, with a short explanation on what exactly they are. It is important to beforehand realize what metadata is important to include in the application, which is what I will do today. In the table below all variables that I have chosen are shown. I will further discuss these with the teachers whether this is a good selection.

| Variable               | Options                                                                                                                       | Description                                                                                                                                                                               |
|------------------------|-------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Gene                   | 19192 different genes                                                                                                         | Allows the user to filter on gene, which allows visualisation of the expression per cell line of the chosen gene.                                                                         |
| Age category           | adult, pediatric, fetus, unknown                                                                                              | Allows the user to filter on age category, which is useful in research in cancer types in specific life stages.                                                                           |
| Sex                    | female, male, unknown                                                                                                         | Allows the user to filter on sex, which is useful if they are interested in cancer cell lines in specifically females or males.                                                           |
| Race                   | caucasian, asian, african, east_indian, north_african, black_or_african_american, american_indian_or_native_american, unknown | Allows the user to filter on race, which is useful if they are interested in cancer cell lines in specific races.                                                                         |
| OncotreeLineage        | various cancer families, e.g.: lung, kidney, breast, skin                                                                     | This is the cancer 'family', also known as the lineage. Including this filter will allow the user to search within specific fields for cancer, which allows for more generalized results. |
| OncotreePrimaryDisease | various cancer types, e.g.: melanoma, embryonal tumor, retinoblastoma, hepatoblastoma                                         | This is the actual cancer type of the model. Including this filter allows the user to very specifically search for certain types of cancer.                                               |

All of these variables had a very low amount to no missing values, as can be seen at the '*Visualising missing data*' chapter.


## Developing a wireframe

**21-11-2024**

Before starting on making the application, it is good to think of a general design first. Today I will do this using a [wireframe builder website](https://wireframe.cc/). My idea for the application design is to have all the user parameters on the left side and have a display for the generated figure on the right side. The left side first consists of a short explanation for usage of the tool and background information on what the tool does. The dropdown filters are objects where the user can choose a specific filter on a specific variable, e.g. selecting "Adult" for the filter "Age Category". For the slider objects, the user can give a certain value for certain variables, such as 'minimum expression levels' or 'number of cell lines displayed'. I want to make the graph scrollable, if possible, for when a user wants to view a large number of cell lines for the chosen gene. However I will have to look into whether a limit should be set for this, as to not make the application lag too much. At the bottom of the left side there are two checkbox objects. Here the user can check/uncheck certain filters, such as 'display cell lines where expression = 0'. Above the generated figure, the chosen parameters will be updated and displayed each time a new figure is generated.


The variables I want to include in the application have been discussed in the former subchapter. 

![A concept wireframe for the DepMap data visualisation application, made using wireframe.cc](../media/wireframe.png)

**02-12-2024**

Today I have decided to add a second tab to the application, in which the filtered data is showcased in a table. So the user can easily view the values of the data, instead of only being able to read them from a graph. For this tab I have made another wireframe (see figure):

![A concept wireframe for the table tab of the application, made using wireframe.cc](../media/wireframe2.png)

**Update 21-01-2024**

In the end, I ended up with a 'merged' version of my two concept wireframes. I kept the layout roughly the same, but I changed to working with cards within one screen, to truly get the dashboard feel.


**21-11-2024**

I have started trying to develop the application using the documentation for Shiny and using the online version of the book ('Mastering Shiny')[https://mastering-shiny.org/index.html] by Hadley Wickham. This is by no means the final version of the application, I will be using the wire frame I have made to start a rough sketch of the application and will fine tune it later, by asking for feedback. 

For today I just focused on making a basic layout, according to the wire frame. First I will be adding the code for a basic ui layout, as following:

```
ui <- fluidPage(
  titlePanel(),
  sidebarLayout(
    sidebarPanel(),
    mainPanel()
  )
)
```
Next I add the actual elements that the user can interact with, which are `selectInput` objects. I do this for each of the parameters that I want the user to be able to change. Below is an example for the selectInput for the cancer type. By adding selected = TRUE, the aplpication will automatically select the option that is at the top of the list. By adding the sort()-function, the choices will be displayed in alphabetical order from A-Z.

```{r}
selectInput("onco_type", label = "Select type of cancer", selected = TRUE, choices =
                    sort(merged$OncotreePrimaryDisease))
```

I also add a `submit-button`, so that when I add the plot, the application won't try to constantly refresh the plot every time the user selects an option, because there is 5 different parameters which means that if the user wants to change those 5 parameters to generate 1 plot, the application will refresh the plot 5 times in the meantime. By adding the submit button the user can decide when they want to refresh and show a plot with the newly adjusted parameters.

```{r}
submitButton(text = "Apply settings")
```

**24-11-2024**

The current 'selectInputs' with the dropdown menus for all parameters offered so far (gene_name, onco_type, sex, race, and age_category) all offer a lot of choices, with gene names for example being over 19000 options. When running the application I would get warnings on each of these selectInputs having too many options and this was noticeable by the long time the app took to launch every time. Today I looked into the problem and I discovered a better alternative to the selectInputs function, which is the 'selectizeInput' function. `selectizeInput` can be used in 2 different ways. 

The first way is to just replace all the selectInputs in the ui, however this will not solve the problem of the application being slow, as this uses the client-side selectize input. The client-side selectize input uses JavaScript to process the searching as the user is typing in an option. So for example when a user starts typing 'Fe' in the sex input, the application will show 'Female' as an option. It works the same with other inputs with more choices, such as the gene_name. 

The second way to use `selectizeInput`, is to add it to the ui and update it through the server, this is the `server-side selectize input`. The server-side selectize input uses R to process the searching, which (when updated) returns the filtered data to selectize. The latter option is way more efficient and thus a good option for solving the problem I have with the application launching and updating too slow.

I have updated all my `selectInput` functions to `selectizeInput` and have added a selectizeInput updating function to the server, which updates all of the selectizeInputs from the ui.

From here I on I will focus on programming the actual app itself and will not further be documenting my steps.



# Appendix

| **Column Name**                  | **Description**                                                                                             |
|----------------------------------|---------------------------------------------------------------------------------------------------------|
| **ModelID**                          | Unique identifier for the model                                                                          |
| **PatientID**                        | Unique identifier for models derived from the same tissue sample                                          |
| **CellLineName**                     | Commonly used cell line name                                                                             |
| **StrippedCellLineName**             | Commonly used cell line name without characters or spaces                                                |
| **DepmapModelType**                  | Abbreviated ID for model type. For cancer models, this field is from Oncotree, information for other disease types are generated by DepMap |
| **OncotreeLineage**                  | Lineage of model. For cancer models, this field is from Oncotree, information for other disease types are generated by DepMap        |
| **OncotreePrimaryDisease**           | Primary disease of model. For cancer models, this field is from Oncotree, information for other disease types are generated by DepMap  |
| **OncotreeSubtype**                  | Subtype of model. For cancer models, this field is from Oncotree, information for other disease types are generated by DepMap         |
| **OncotreeCode**                     | For cancer models, this field is based on Oncotree. For some models for which no corresponding code exists, this field is left blank     |
| **PatientSubtypeFeatures**           | Aggregated features known for the patient tumor                                                         |
| **RRID**                             | Cellosaurus ID                                                                                          |
| **Age**                              | Age at time of sampling                                                                                 |
| **AgeCategory**                      | Age category at time of sampling (Adult, Pediatric, Fetus, Unknown)                                      |
| **Sex**                              | Sex at time of sampling (Female, Male, Unknown)                                                         |
| **PatientRace**                      | Patient/clinical indicated race (not derived)                                                           |
| **PrimaryOrMetastasis**              | Site of the primary tumor (Primary, Metastatic, Recurrence, Other, Unknown)                             |
| **SampleCollectionSite**             | Site of tissue sample collection                                                                         |
| **SourceType**                       | Indicates where model was onboarded from (Commercial, Academic lab, Other)                              |
| **SourceDetail**                     | Details on where model was onboarded from                                                                |
| **CatalogNumber**                    | Catalog number of cell model, if commercial                                                              |
| **ModelType**                        | Type of model at onboarding (e.g., Organoid, Cell Line)                                                 |
| **TissueOrigin**                     | Indicates tissue model was derived from (Human, Mouse, Other)                                           |
| **ModelDerivationMaterial**          | Indicates what material a model was derived from (Fresh tissue, PDX, Other)                             |
| **ModelTreatment**                   | Indicates which virus was used to transform a cell line                                                 |
| **PatientTreatmentStatus**           | Indicates if sample was collected before, during, or after the patient's cancer treatment (Pre-treatment, Active treatment, Post-treatment, Unknown)|
| **PatientTreatmentType**             | Type of treatment patient received prior to, or at the time of, sampling (e.g. chemotherapy, immunotherapy, etc.), if known|
| **PatientTreatmentDetails**          | Details about patient treatment                                                                           |
| **Stage**                            | Stage of patient tumor                                                                                  |
| **StagingSystem**                    | Classification system used to categorize disease stage (e.g., AJCC Pathologic Stage), if known           |
| **PatientTumorGrade**                | Grade (or other marker of proliferation) of the patient tumor, if known                                 |
| **PatientTreatmentResponse**         | Any response to treatment, if known                                                                      |
| **GrowthPattern**                    | Format model onboarded in (Adherent, Suspension, Organoid, Neurosphere, Neurosphere 2D, Unknown)        |
| **OnboardedMedia**                   | Description of onboarding media                                                                          |
| **FormulationID**                    | The unique identifier of the onboarding media                                                           |
| **SerumFreeMedia**                   | Indicates a non-serum based media (<1% serum)                                                           |
| **PlateCoating**                     | Coating on plate model onboarded in (Laminin, Matrigel, Collagen, None)                                 |
| **EngineeredModel**                  | Indicates if model was engineered (genetic knockout, genetic knock down, cultured to resistance, other)   |
| **EngineeredModelDetails**           | Detailed information for genetic knockdown/out models                                                    |
| **CulturedResistanceDrug**           | Drug of resistance used for cultured to resistance models                                                |
| **PublicComments**                   | Comments released to portals                                                                              |
| **CCLEName**                         | CCLE name for the cell line                                                                               |
| **HCMIID**                           | Identifier models available through the Human Cancer Models Initiative (HCMI)                           |
| **ModelAvailableInDbgap**            | Indicates the availability of data for a Model on DbGaP. Refer to the “SharedToDbgap” column on OmicsProfile.csv for specific Omics Profile data available|
| **ModelSubtypeFeatures**             | Curated list of confirmed molecular features seen in the model                                           |
| **WTSIMasterCellID**                 | WTSI ID                                                                                                  |
| **SangerModelID**                    | Sanger ID                                                                                               |
| **COSMICID**                         | Cosmic ID 



